homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -249,9 +249,24 @@ def _async_register_button_devices(
         seen.add(ha_device_serial)
 
         area = _area_name_from_id(bridge.areas, ha_device["area"])
-        # name field is still a combination of area and name from pylytron-caseta
-        # extract the name portion only.
-        name = ha_device["name"].split("_")[-1]
+
+        # Build the name of the Keypad/Pico
+        ha_device_name_prefix = ""
+        ha_device_name_suffix = ""
+        if "control_station_name" in ha_device:
+            # RA3/HWQSX Control Pico or Keypad, Assemble the Device Name
+            ha_device_name_prefix = ha_device["control_station_name"]
+            if "Pico" in ha_device["type"]:
+                ha_device_name_suffix = "Pico"
+            elif "Keypad" in ha_device["type"]:
+                ha_device_name_suffix = "Keypad"
+        ha_device_name_parts = [
+            ha_device_name_prefix,
+            ha_device["device_name"],
+            ha_device_name_suffix,
+        ]
+        name = " ".join(filter(None, ha_device_name_parts))
+
```

Comment:
Let's make this its own function.

Also I think you can write the `Pico` and `Keypad` into a loop.  Ideally we come up with a way to avoid using the raw strings

Reply:
As this is all getting a bit complex it might make sense to make a class to keep track of it all and only generate the data once and reuse it in various forms